_im-plugin/ism/api.md

```diff
@@ -546,3 +546,154 @@ DELETE _plugins/_ism/policies/policy_1
   "_primary_term": 1
 }
 ```
+
+## Error prevention validation
+Introduced 2.4
+{: .label .label-purple }
+
+ISM allows you to execute an action automatically. However, action execution can fail for a variety of reasons. Error prevention validation checks the possible reason for a failure before execution of an action.
+
+To enable error prevention validation, set the `plugins.index_state_management.validation_service.enabled` setting to `true`:
+
+#### Request
+
+```bash
+PUT _cluster/settings
+{
+   "persistent":{
+      "plugins.index_state_management.validation_action.enabled": true
+   }
+}
+```
+
+#### Sample response
+
+```json
+{
+  "acknowledged" : true,
+  "persistent" : {
+    "plugins" : {
+      "index_state_management" : {
+        "validation_action" : {
+          "enabled" : "true"
+        }
+      }
+    }
+  },
+  "transient" : { }
+}
+```
+
+To check an error prevention validation status and message, pass `validate_action=true` to the `_plugins/_ism/explain` endpoint:
+
+#### Request
+
+```bash
+GET _plugins/_ism/explain/test-000001?validate_action=true
+```
+
+#### Sample Response
```

Comment:
Something like "The response contains an additional `validate` object with a validation message and status: